### PR TITLE
soc: atmel_sam0: Add SAMD5x/SAME5x family

### PR DESCRIPTION
**Pull request by benpicco**
The SAME5x/SAMD5x is a line of Cortex-M4F MCUs that share peripherals with the sam0 Cortex-M0+ and saml1x Cortex-M23 parts.

split off from #14685

### CODE REVIEW

File: dts/arm/atmel/samd5x.dtsi [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4";
+			reg = <0>;
+		};
+	};
+
+	sram0: memory@20000000 {
+		device_type = "memory";
```
Comment (galak) [inline dts/arm/atmel/samd5x.dtsi]: remove `device_type`, its not needed for on chip sram.

Also move this node inside the SoC node.

File: dts/arm/atmel/samd5x.dtsi [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+		compatible = "mmio-sram";
+		reg = <0x20000000 0x40000>;
+	};
+
+	backup0: memory@47000000 {
```
Comment (galak) [inline dts/arm/atmel/samd5x.dtsi]: I assume memory@4700000, flash, and device_id are all part of the SoC.  If such please move these nodes inside the SoC container node.

File: soc/arm/atmel_sam0/samd51/dts_fixup.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ * Copyright (c) 2019 ML!PA Consulting GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* SoC level DTS fixup file */
+
+#define DT_FLASH_DEV_NAME DT_ATMEL_SAM0_NVMCTRL_0_LABEL
+
+#define CONFIG_WDT_0_NAME DT_ATMEL_SAM0_WATCHDOG_0_LABEL
```
Comment (galak) [inline soc/arm/atmel_sam0/samd51/dts_fixup.h]: CONFIG_WDT_0_NAME can be removed now.

File: soc/arm/atmel_sam0/same51/dts_fixup.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ * Copyright (c) 2019 ML!PA Consulting GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* SoC level DTS fixup file */
+
+#define DT_FLASH_DEV_NAME DT_ATMEL_SAM0_NVMCTRL_0_LABEL
+
+#define CONFIG_WDT_0_NAME DT_ATMEL_SAM0_WATCHDOG_0_LABEL
```
Comment (galak) [inline soc/arm/atmel_sam0/same51/dts_fixup.h]: remove CONFIG_WDT_0_NAME

File: soc/arm/atmel_sam0/same53/dts_fixup.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+ * Copyright (c) 2019 ML!PA Consulting GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* SoC level DTS fixup file */
+
+#define DT_FLASH_DEV_NAME DT_ATMEL_SAM0_NVMCTRL_0_LABEL
+
+#define CONFIG_WDT_0_NAME DT_ATMEL_SAM0_WATCHDOG_0_LABEL
```
Comment (galak) [inline soc/arm/atmel_sam0/same53/dts_fixup.h]: remove CONFIG_WDT_0_NAME

File: soc/arm/atmel_sam0/same54/dts_fixup.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+ * Copyright (c) 2019 ML!PA Consulting GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* SoC level DTS fixup file */
+
+#define DT_FLASH_DEV_NAME DT_ATMEL_SAM0_NVMCTRL_0_LABEL
+
+#define CONFIG_WDT_0_NAME DT_ATMEL_SAM0_WATCHDOG_0_LABEL
```
Comment (galak) [inline soc/arm/atmel_sam0/same54/dts_fixup.h]: Remove CONFIG_WDT_0_NAME

**Review by galak — CHANGES_REQUESTED** [review]
Some DT related cleanups but otherwise looks good.

File: dts/arm/atmel/samd5x.dtsi (hunk 2 above)
Comment (benpicco) [reply dts/arm/atmel/samd5x.dtsi]: Alright, moved them.

File: soc/arm/atmel_sam0/same51/dts_fixup.h (hunk 4 above)
Comment (benpicco) [reply soc/arm/atmel_sam0/same51/dts_fixup.h]: And it's gone.

**Review by galak — APPROVED** [review]

**Pull request merged**